java/matcher_integration_test.go

```diff
@@ -0,0 +1,95 @@
+package java
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/quay/claircore"
+	internalMatcher "github.com/quay/claircore/internal/matcher"
+	"github.com/quay/zlog"
+
+	"github.com/quay/claircore/datastore/postgres"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/libvuln/updates"
+	"github.com/quay/claircore/pkg/ctxlock"
+	"github.com/quay/claircore/test/integration"
+	pgtest "github.com/quay/claircore/test/postgres"
+	"github.com/quay/claircore/updater/osv"
+)
+
+func TestMain(m *testing.M) {
+	var c int
+	defer func() { os.Exit(c) }()
+	defer integration.DBSetup()()
+	c = m.Run()
+}
+
+func TestMatcherIntegration(t *testing.T) {
+	integration.NeedDB(t)
+	ctx := zlog.Test(context.Background(), t)
+	pool := pgtest.TestMatcherDB(ctx, t)
+	store := postgres.NewMatcherStore(pool)
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusTeapot)
+	}))
+	defer srv.Close()
+
+	m := &matcher{}
+	locks, err := ctxlock.New(ctx, pool)
+	if err != nil {
+		t.Error(err)
```

Comment:
> I don't recall. Will the test still continue after this is called? Should the test not be able to continue if this is reached?

I think it can be changed to `t.Fatalf("%v", err)` ? @hdonnay